lib/pos_mempool.go

```diff
@@ -241,11 +270,70 @@ func (mp *PosMempool) Start() error {
 			return errors.Wrapf(err, "PosMempool.Start: Problem loading persisted transactions")
 		}
 	}
+	mp.startAugmentedViewRefreshRoutine()
 
 	mp.status = PosMempoolStatusRunning
 	return nil
 }
 
+func (mp *PosMempool) startAugmentedViewRefreshRoutine() {
+	go func() {
+		for {
+			select {
+			case <-time.After(time.Duration(mp.augmentedBlockViewRefreshIntervalMillis) * time.Millisecond):
+				// If we're not within 10 blocks of the PoS cutover, we don't need to update the
+				// augmentedLatestBlockView.
+				if mp.latestBlockHeight+10 < uint64(mp.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+					continue
+				}
+				// Update the augmentedLatestBlockView with the latest block view.
+				mp.RLock()
+				newView, err := mp.readOnlyLatestBlockView.CopyUtxoView()
+				if err != nil {
+					glog.Errorf("PosMempool.startAugmentedViewRefreshRoutine: Problem copying utxo view: %v", err)
+					continue
+				}
+				mp.RUnlock()
+				for _, txn := range mp.GetTransactions() {
+					copiedView, err := newView.CopyUtxoView()
+					if err != nil {
+						glog.Errorf("PosMempool.startAugmentedViewRefreshRoutine: Problem copying utxo view: %v", err)
+						continue
+					}
+					_, _, _, _, err = copiedView.ConnectTransaction(
+						txn.GetTxn(), txn.Hash(), uint32(mp.latestBlockHeight)+1, time.Now().UnixNano(), false,
+						false)
+					// If the transaction successfully connects, we set the newView to the copiedView
+					// and proceed to the next transaction.
+					if err == nil {
+						newView = copiedView
+						continue
+					}
+					// If the transaction failed to connect, we connect the transaction as a failed txn
+					// directly on newView.
+					if mp.latestBlockHeight+1 >= uint64(mp.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+						// Try to connect as failing txn directly to newView
+						_, _, _, err = newView._connectFailingTransaction(
+							txn.GetTxn(), uint32(mp.latestBlockHeight), false)
+						if err != nil {
+							glog.Errorf(
```

Comment:
I think if this fails, then newView is compromised and you have to recompute it before you go to the next iteration. Maybe what I would do to keep it light is set `copiedView, err = newView.CopyUtxoView()` and run this check on copiedView to guarantee newView is always clean. And then if it passes you do `newView = copiedView` like above.

Reply:
Update: Resolved this myself.